### ISSUE TITLE
Implement DependencyInjector

### ISSUE THREAD

**Issue by Gnat008** [Gnat008/PerWorldInventory#62]
Hi @ljacqu ! xD

Anyway. Implement new dependency injector. This will make creating new classes and functionality easier and helps with clearer code. Removes the necessity of having to call massive constructors for every class, yet still giving them access to everything they need to function.

Additionally, writing unit tests will me much easier or actually possible compared to how the project is set up currently.

Link for anyone not familiar with this (and to remind myself where it is): https://github.com/ljacqu/DependencyInjector

**Comment by ljacqu** [comment]
Thanks for the mention :smile:
I still need to find how to put it on a Maven repo. Right now that's all that's left for a first release. ;)

**Comment by Gnat008** [comment]
@ljacqu Not sure if this helps or not, but I'll drop the link here anyway: http://www.theserverside.com/news/1364121/Setting-Up-a-Maven-Repository

**Comment by ljacqu** [comment]
Thanks! Currently in the process of setting it up to be available from Maven Central. Quite a lot of things to set up, though.

**Comment by ljacqu** [comment]
It is now available from Maven Central. :)

```xml
<dependency>
    <groupId>ch.jalu</groupId>
    <artifactId>injector</artifactId>
    <version>0.1</version>
</dependency>
```

**Comment by ljacqu** [comment]
@Gnat008, 
Released 0.2 which allows to turn off method or field scanning for a class by annotating it with `@NoMethodScan` or `@NoFieldScan`.

```xml
<dependency>
    <groupId>ch.jalu</groupId>
    <artifactId>injector</artifactId>
    <version>0.2</version>
</dependency>
```
